Resources/MessageQueue/StompMQConnector.py

```diff
@@ -46,53 +55,55 @@ def setupConnection(self, parameters=None):
     if not all(p in parameters for p in ('Host', 'VHost')):
       return S_ERROR('Input parameters are missing!')
 
-    # Make the actual connection
+    reconnectSleepInitial = self.parameters.get('ReconnectSleepInitial', StompMQConnector.RECONNECT_SLEEP_INITIAL)
+    reconnectSleepIncrease = self.parameters.get('ReconnectSleepIncrease', StompMQConnector.RECONNECT_SLEEP_INCREASE)
+    reconnectSleepMax = self.parameters.get('ReconnectSleepMax', StompMQConnector.RECONNECT_SLEEP_MAX)
+    reconnectSleepJitter = self.parameters.get('ReconnectSleepJitter', StompMQConnector.RECONNECT_SLEEP_JITTER)
+    reconnectAttemptsMax = self.parameters.get('ReconnectAttemptsMax', StompMQConnector.RECONNECT_ATTEMPTS_MAX)
+
     host = self.parameters.get('Host')
-    port = self.parameters.get('Port', 61613)
+    port = self.parameters.get('Port', StompMQConnector.PORT)
     vhost = self.parameters.get('VHost')
 
     sslVersion = self.parameters.get('SSLVersion')
     hostcert = self.parameters.get('HostCertificate')
     hostkey = self.parameters.get('HostKey')
-    # get local key and certificate if not available via configuration
-    if sslVersion and not (hostcert or hostkey):
-      paths = Locations.getHostCertificateAndKeyLocation()
-      if not paths:
-        return S_ERROR('Could not find a certificate!')
-      else:
-        hostcert = paths[0]
-        hostkey = paths[1]
-
-    try:
 
-      # get IP addresses of brokers
-      brokers = socket.gethostbyname_ex(host)
-      self.log.info('Broker name resolves to %s IP(s)' % len(brokers[2]))
+    connectionArgs = {'vhost': vhost,
+                      'keepalive': True,
+                      'reconnect_sleep_initial': reconnectSleepInitial,
+                      'reconnect_sleep_increase': reconnectSleepIncrease,
+                      'reconnect_sleep_max': reconnectSleepMax,
+                      'reconnect_sleep_jitter': reconnectSleepJitter,
+                      'reconnect_attempts_max': reconnectAttemptsMax}
 
-      if sslVersion is None:
-        pass
-      elif sslVersion == 'TLSv1':
+    if sslVersion is not None:
```

Comment:
what happens if sslVersion is something different from `TLSv1` ?

Reply:
Then  it should return:
S_ERROR(EMQCONN, 'Invalid SSL version provided: %s' % sslVersion)

Reply:
yes but that's not what will happen. If you have `sslVersion = toto`, it will pass the first test because not `None`, and  not enter the second contiond, and nothing will happen. Your first test should just be replaced by the second one altogether I suppose